### ISSUE TITLE
JavadocTagContinuationIndentation: add strict enforcement of indentation

### ISSUE THREAD

**Issue by rnveach** [checkstyle/checkstyle#8182]
Split from https://github.com/checkstyle/checkstyle/issues/2813#issue-125702063 ,

https://checkstyle.org/config_javadoc.html#JavadocTagContinuationIndentation

TestClass.java

```
package test;

public class TestClass {
    /**
     * Desc.
     * 
     * @param s
     * Desc.
     * Desc 2.
     * Desc 3.
     *                         Desc 4. Line 11. violation is expected
     */
    public void test(String s) {
    }
}
```

config.xml

```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="ASCII" />
    <property name="severity" value="error" />
    <property name="fileExtensions" value="java" />

    <module name="TreeWalker">
        <module name="JavadocTagContinuationIndentation" />
    </module>
</module>
```

output from: `java -jar checkstyle-6.14.1-all.jar -c config.xml TestClass.java`

```
Starting audit...
[ERROR] TestClass.java:9: Line continuation have incorrect indentation level, 
    expected level should be 4. [JavadocTagContinuationIndentation]
[ERROR] TestClass.java:10: Line continuation have incorrect indentation level, 
     expected level should be 4. [JavadocTagContinuationIndentation]
Audit done.
Checkstyle ends with 2 errors.
```

Expected **extra** output:

```
[ERROR] TestClass.java:8: Line continuation have incorrect indentation level, 
    expected level should be 4. [JavadocTagContinuationIndentation]
[ERROR] TestClass.java:11: Line continuation have incorrect indentation level,
     expected level should be 4. [JavadocTagContinuationIndentation]
```

Line 8 is covered in https://github.com/checkstyle/checkstyle/issues/2813#issue-125702063 , this issue is about line 11 only.

I would like Line 11 to be required to be at the same position as previous indented lines, but it seems `JavadocTagContinuationIndentation` ignores it once it is indented past its minimum, required position.

We should create a new property called `forceStrictCondition` ( same is in https://checkstyle.org/config_misc.html#Indentation_Properties ) with a default value of false to not conflict with google style.

**Comment by Gaurav-Punjabi** [comment]
I'm on it, once https://github.com/checkstyle/checkstyle/pull/8002 is merged. As it will change the core logic.

**Comment by romani** [comment]
here is abandoned PR https://github.com/checkstyle/checkstyle/pull/8194 , some work from it might be reused to create new PR .
everyone is welcome with PR.

**Comment by mohitsatr** [comment]
Im on it

**Comment by romani** [comment]
@mohitsatr , please confirm that this is how google formatter is working to update code.
Please quote style guide to let me confirm that it is what style guide is requesting.

I am interested why we think it is google style issue.

**Comment by mohitsatr** [comment]
@romani It is not google-style issue. The Pr was send before the project the started.
